Masquerade4997Code/src/main/java/org/firstinspires/ftc/teamcode/Robots/MarkOne/TeleOp/RobotTeleOp.java

```diff
@@ -61,13 +59,11 @@ public void runLinearOpMode() throws InterruptedException {
             else if (controller2.leftTriggerPressed()) robot.lift.setVelocity(-controller2.leftTrigger());
             else robot.lift.setVelocity(0);
 
-            if (robot.lift.encoder.getInches() > 24) controller2.toggle(controller2.yOnPress(), robot.blockRotater, prevRotater);
-            controller2.toggle(controller2.xOnPress(), robot.blockGrabber, prevGrabber);
-            controller2.toggle(controller2.aOnPress(), robot.blockPusher,prevPusher);
-            if (controller1.b()) robot.lowerFoundationHook();
-            else if (controller1.x()) robot.raiseFoundationHook();
-            else robot.midFoundationHook();
-            controller1.toggle(controller1.xOnPress(), robot.sideGrabber, prevSide);
+            if (robot.lift.encoder.getInches() > 24) MasqUtils.toggle(controller2.yOnPress(), robot.blockRotater, prevRotater);
+            MasqUtils.toggle(controller2.xOnPress(), robot.blockGrabber, prevGrabber);
+            MasqUtils.toggle(controller2.aOnPress(), robot.blockPusher,prevPusher);
+            robot.foundationHook.DriverControl(controller1);
```

Comment:
Put all driverControls at the bottom of the while loop. They are unimportant in normal flow so I usually just ut em all together at the end.

Reply:
That is the end all of the code is driver control and I can’t put the controller resets, dash updates and previous values before them.

Reply:
The next line of code is a MasqUtils.toggle

Reply:
Which is driver control

Reply:
The next line is 
`MasqUtils.toggle(controller2.rightBumper() || controller2.leftBumper(), robot.sideGrabber, prevSide);`
Above `robot.foundationHook.DriverControl(controller1);` you also have other MasqUtils.toggle statements.

Reply:
toggling is moving the servos when the driver presses a button in this case either bumper which because the drivers are controlling the servos it is by definition driver control

Reply:
The line previous to 65 is a toggle. The line after 65 is a toggle. Why can't 65 be put after 66.

Reply:
Okay I will make this change